### ISSUE TITLE
Impossible to use with legacy txs

### ISSUE THREAD

**Issue by tynes** [bcoin-org/bledger#20]
On this line:

https://github.com/bcoin-org/bledger/blob/d4364b6b651f154f74aadcc7f931ebe0c7f4a6b3/lib/txinput.js#L94

If `options.witness` is false, then it asserts`!this.isWitness()`
which eventually tries to look at `this.coin`, which is still assigned a
`null` value in the constructor, and is not assigned anything until strictly
afterwards in the control flow, 

https://github.com/bcoin-org/bledger/blob/d4364b6b651f154f74aadcc7f931ebe0c7f4a6b3/lib/txinput.js#L113

so this code path appears will always throw
an error as `this.getCoin` returns `null` and has no property `script`

```js
  getPrev() {
    if (!this._prev)
      this._prev = this.getCoin().script;

    return this._prev;
  }
```

**Comment by tynes** [comment]
Maybe we can make `this.tx` and `this.index` required and then assign `this.coin` no matter what, then we can assert `this.isWitness()` if `options.witness` is `true` and `!this.isWitness()` if `options.witness` is `false`

**Comment by nodar-chkuaselidze** [comment]
Fixed 99460b0b

**Comment by nodar-chkuaselidze** [comment]
@tynes is this still an issue ?
